lib/notifications/db/notification.ex

```diff
@@ -21,6 +21,7 @@ defmodule Notifications.Db.Notification do
     field(:service_id, :string)
     field(:start_time, :integer)
     field(:end_time, :integer)
+    field(:state, :string, virtual: true)
```

Comment:
Maybe I'm missing something, but what's `state` on `Notification` (as opposed to `NotificationUser`) for? The places I'm seeing so far reference the field on `NotificationUser`, which is what I'd expect.

Reply:
@lemald see the SQL query in `Notifications.Notification.unexpired_notifications_for_user`. Basically, this is the only way I could find to get Ecto to generate the SQL I want, which is something like:

`SELECT notifications.*, notifications_users.state FROM notifications INNER JOIN notifications_users ON notifications.id = notifications_users.notification_id WHERE [some stuff]`

The alternative seemed to involve spelling out every column I wanted from `notifications` (i.e. all of them), which seemed both inelegant and brittle.

Reply:
Ah, okay. Yeah, I don't have any good theories for other ways to accomplish that, and manually specifying ever column would be bad. As long as this field doesn't cause problems for other queries that might not join to `NotificationUser` (which I don't think it will?) I think this is fine.